src/autiter.rs

```diff
@@ -53,13 +53,13 @@ pub trait Automaton<P>: Sized {
     }
 
     /// Returns an iterator of non-overlapping matches in `s`.
-    fn find<'a, 's>(
+    fn find<'a, 's, Q: ?Sized + AsRef<[u8]>>(
         &'a self,
-        s: &'s str,
+        s: &'s Q,
     ) -> Matches<'a, 's, P, Self> {
```

Comment:
@jneem Does adding `where Self: Sized` work here? I think it should, which should make it object safe. (Similarly for `find_overlapping`.)